packages/server/src/script/loadEmailTesting.ts

```diff
@@ -0,0 +1,127 @@
+import { sampleStateInfo } from '../service/letter'
+import { contactRecords } from '../service/contact'
+import { makeClient, httpConnector } from '@tianhuil/simple-trpc/dist/client'
+import { IVbmRpc, processEnvOrThrow } from '../common'
+import fetch from 'node-fetch'
+
+const serverUrl = processEnvOrThrow('REACT_APP_SERVER')
+const timeout = parseInt(processEnvOrThrow('REACT_APP_TIMEOUT'))
+
+export const client = makeClient<IVbmRpc>(
+  httpConnector(serverUrl, { timeout, fetch })
+)
+
+const fakeEmail = 'fake_voter@votebymail.io'
+
+interface Result {
+  success: number
+  failures: number
+}
+
+/**
+ * Asynchronously waits for a array full of promises to build, the promises
+ * inside the array might not be fulfilled yet--this function only makes sure
+ * the array is filled to the desired size.
+ *
+ * @param size The size of the array
+ * @param promise Function returning a promises to fill the array
+ * @param interval The amount of time in ms before pushing the next item
+ */
+const fillPromiseArray = async <T>(size: number, promise: () => Promise<T>, interval?: number): Promise<Promise<T | void>[]> => {
+  const promises: Promise<T | void>[] = []
+
+  // Not using Array.fill because the promises array can be filled based
+  // on an interval--and we want to manually resolve the 'filledArray' promise
+  // below.
+  await new Promise(filledArray => {
+    const loop = setInterval(
+      () => {
+        promises.push(promise())
+        const length = promises.length
+        if (length === size) {
```

Comment:
This is a very weird way of doing it.  Because of a race condition, you could end up with more promises started than `size`.  Instead, just use `setTimeout` n times.

Also, the results could come in out of order if you use `promises.push`.  Instead, use the index of the promise to set the array position.